examples/shopping_cart.rs

```diff
@@ -0,0 +1,72 @@
+#![feature(type_alias_enum_variants)]
+
+#[macro_use]
+extern crate mirai_annotations;
+
+pub mod foreign_contracts;
+
+struct Cart {
+    items: Vec<i32>
+}
+
+impl Cart {
+    fn new() -> Cart {
+        Cart {
+            items: Vec::new(),
+        }
+    }
+
+    fn add_item(&mut self, value: i32) {
+        precondition!(value > 0);
+        let old_len = self.items.len();
+        self.items.push(value);
+        verify!(self.items.len() == old_len + 1);
```

Comment:
This just verifies that push has the obvious post condition. This is not a condition we would expect in real code.